### PR TITLE
Upgrade to kubebuilder v3

### PR DESCRIPTION
**Pull request by ptnapoleon**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- Upgrades this project to rely on kubebuilder v3 and related dependencies, from kubebuilder v2
- Changes spec.DurationSeconds to spec.Duration that takes a duration string, rather than an integer

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

I still have to add documentation for changes, as well as test out possible new features of kubebuilder v3 that may be worth including immediately

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -118,11 +118,11 @@ func main() {
 	// parse flags
 	pflag.StringVar(&configPath, "config", "", "Configuration file path")
 
-	pflag.StringVar(&cfg.Controller.MetricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
-	handleFatalError(viper.BindPFlag("controller.metrics.addr", pflag.Lookup("metrics-addr")))
+	pflag.StringVar(&cfg.Controller.MetricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
```
Comment (Azoam) [inline main.go:121]: Just a nitpick, `.MetricsAddr` and `metrics-bind-address`, Maybe we should change to `.MetricsBindAddr`

**Review by Azoam — APPROVED** [review]
LGTM

File: main.go (hunk 1 above)
Comment (ptnapoleon) [reply main.go:121]: Great suggestion

File: .circleci/config.yml [hunk 2]
```diff
@@ -45,9 +45,9 @@ executors:
     docker:
       # TODO: Publicize the chaos controller ci image to the datadog docker registry
       #- image: datadog/chaos-controller-runner-circle:<< pipeline.parameters.CURRENT_CI_IMAGE >>
-      - image: golang:1.15-buster
+      - image: golang:1.16-buster
```
[the remark this answers is not in the document]
Comment (ptnapoleon) [reply .circleci/config.yml:48]: I couldn't find the 1.17 image

File: api/v1beta1/disruption_webhook.go [hunk 3]
```diff
@@ -23,20 +24,38 @@ var k8sClient client.Client
...
 		Complete()
 }
 
+//+kubebuilder:webhook:webhookVersions={v1beta1},path=/mutate-chaos-datadoghq-com-v1beta1-disruption,mutating=true,failurePolicy=fail,sideEffects=None,groups=chaos.datadoghq.com,resources=disruptions,verbs=create;update,versions=v1beta1,name=mdisruption.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Defaulter = &Disruption{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Disruption) Default() {
+	logger.Infow("BUGS!", "name", r.Name, "d", r.Spec.Duration, "dd", r.Spec.Duration.Duration())
```
[the remark this answers is not in the document]
Comment (ptnapoleon) [reply api/v1beta1/disruption_webhook.go:49]: Definitely not

File: chart/install.yaml [hunk 4]
```diff
@@ -888,3 +864,35 @@ webhooks:
     - DELETE
     resources:
     - disruptions
+---
+# Source: chaos-controller/templates/webhook.yaml
```
[the remark this answers is not in the document]
Comment (ptnapoleon) [reply chart/install.yaml:868]: Yes

**Comment by ptnapoleon** [comment]
Tests are de-flakied and fixes have been addressed

**Review by Azoam — APPROVED** [review]

**Review by Devatoria — APPROVED** [review]

**Pull request merged**